src/sys/linux/os.rs

```diff
@@ -1,5 +1,25 @@
 use ctypes::c_char;
+use error::Error as StdError;
+use ffi::{CStr, CString, OsStr, OsString};
+use fmt;
+use io;
+use iter;
+use libc;
 use linux;
+use marker::PhantomData;
+use path::{self, PathBuf};
+use slice;
+use super::cvt;
+use sys::ext::prelude::*;
+use vec;
+
+static ENV_LOCK: () = ();
+// TODO(steed): Synchronize environment access once we have mutexes.
```

Comment:
Could you open an issue about this? Now that we have threads this one may go very wrong if two or more threads try to `set_env` the same variable :-/.